modules/test/playwright/tests/dynamic-data-mapping-form-web/formFields.spec.ts

```diff
@@ -187,6 +187,63 @@ test.describe('Manage fields through Form Preview page', () => {
 		await newTabPage.close();
 	});
 
+	test('verify boolean field aria-labelledby have corresponding element', async ({
+		formBuilderPage,
+		formBuilderSidePanelPage,
+	}) => {
+		await formBuilderPage.goToNew();
+
+		await formBuilderPage.fillFormTitle('Form' + getRandomInt());
+
+		await formBuilderSidePanelPage.addFieldByDoubleClick('Boolean');
+
+		await formBuilderSidePanelPage.label.fill('Boolean without helptext');
+
+		await formBuilderSidePanelPage.backButton.click();
+
+		await formBuilderSidePanelPage.addFieldByDoubleClick('Boolean');
+
+		await formBuilderSidePanelPage.label.fill('Boolean with helptext');
+
+		await formBuilderSidePanelPage.helpText.fill('Help text');
+
+		await formBuilderSidePanelPage.backButton.click();
+
+		const newTabPagePromise = new Promise<Page>((resolve) =>
+			formBuilderPage.page.once('popup', resolve)
+		);
+
+		await formBuilderPage.previewButton.click();
+
+		const newTabPage = await newTabPagePromise;
+
+		await newTabPage.waitForLoadState('domcontentloaded');
+
+		const elementWithoutHelpTextLabel = await newTabPage
+			.locator(
+				'div[aria-labelledby]:has(span.custom-control-label:has-text("Boolean without helptext"))'
+			)
+			.last()
+			.getAttribute('aria-labelledby');
```

Comment:
We talked in person about how we can try to improve this. We can come together to do it together tomorrow and send it forward. 😄